enhancements/installer/aws-cross-account-dns-zone.md

```diff
@@ -162,9 +160,55 @@ platform:
     hostedZone: Z00147933I3NWOQ6M4699
     hostedZoneRole: arn:aws:iam::<account-a>:role/<role-name>
 ```
-Please see Open Questions for further discussion of the install config.
 
+### API: ExternalDNS Operator
+
+[ExternalDNS Operator](https://github.com/openshift/external-dns-operator) allows you to deploy and manage [ExternalDNS](https://github.com/kubernetes-sigs/external-dns),
+a cluster-internal component which makes Kubernetes resources discoverable through public DNS servers. This enhancement
+extends the ExternalDNS Operator to support cross-account DNS zones in AWS. We extend the`ExternalDNS` API object by adding
+the field `RoleARN` to `ExternalDNSAWSProviderOptions` struct (`spec.provider.aws`).
+Refer to [`externaldns_types.go`](https://github.com/openshift/external-dns-operator/blob/main/api/v1beta1/externaldns_types.go)
+for the existing API structure of the `ExternalDNS` object.
+
+In addition to adding `RoleARN`, the `Credentials` field was updated to be optional. Prior to this change, the `Credentials`
+field was required if `spec.provider.aws` was specified. However, OpenShift clusters are exempt from explicitly
+providing `Credentials` since the ExternalDNS Operator generates a `CredentialRequest` automatically. Consequently, to
+enable OpenShift clusters to specify `RoleARN` without `Credentials`, it is necessary to make the `Credentials` field
+optional. The validation webhook takes care of guaranteeing that non-OpenShift clusters include credentials through the
+API.
```

Comment:
So on OpenShift you can have empty `aws` field then:
```yaml
provider:
  type: aws
  aws: {}
```

Reply:
Yes I suppose, but `omitempty` is [specified](https://github.com/openshift/external-dns-operator/blob/68bdec5a19686d0ccdb3ed1f657dbfd14809b338/api/v1beta1/externaldns_types.go#L215) on the `aws` field.

Is this problematic?

Reply:
I will not pass the validating webhook due to [the validate credentials logic for AWS](https://github.com/openshift/external-dns-operator/blob/68bdec5a19686d0ccdb3ed1f657dbfd14809b338/api/v1beta1/externaldns_webhook.go#L138-L140).